internal/controller/dragonfly_instance.go

```diff
@@ -340,9 +362,19 @@ func (dfi *DragonflyInstance) getPods(ctx context.Context) (*corev1.PodList, err
 // to the given master instance
 func (dfi *DragonflyInstance) replicaOf(ctx context.Context, pod *corev1.Pod, masterIp string) error {
 	redisClient := redis.NewClient(&redis.Options{
-		Addr: fmt.Sprintf("%s:%d", pod.Status.PodIP, resources.DragonflyAdminPort),
+		Addr:     fmt.Sprintf("%s:%d", pod.Status.PodIP, resources.DragonflyAdminPort),
+		Username: dfi.dfAdminCreds.username,
+		Password: dfi.dfAdminCreds.password,
 	})
 
+	// if admin credentials are defined, set masterauth so slave
+	// can talk to master over an authenticated connection
+	if dfi.dfAdminCreds.password != "" {
+		if err := redisClient.ConfigSet(ctx, "masterauth", dfi.dfAdminCreds.password).Err(); err != nil {
```

Comment:
Ahh, So this is where we set to instances but I don't think setting here makes sense. We should try to pass it during initialisation through args directly.

Reply:
Hmm. I don't think we are even setting this here. we are asking the password to be used but where are we even configuring the master with the admin password?

I think, the current approach looks a bit complicated. I'd rather prefer somehow letting the user give access to the password secret to the Operator and it using it. instead of this. wdyt?

Reply:
Hi @Pothulapati , thanks for getting back to me! :)

Yes, this requires that the admin password provided to the operator via k8s secret match the password provided to the actual dragonfly instances.  I absolutely agree that sharing the secret between the instances and operator would be better, but I don't see a way around this, since secrets can't be shared across k8s namespaces.  

We'd either have to enforce that the operator and dragonfly instances exist in the same namespace, which would require a major change to `dragonfly-operator.yaml`, or require that the users create a duplicate secret in the `dragonfly-operator-system` that matches the instance secret (like I did in this PR).  Do you know of another way to share the k8s secret?

I figured setting the `masterauth` in replica configuration made sense, since it's an implementation detail tied to the configuration of a replica, but I'm open to moving it elsewhere.

Reply:
> Do you know of another way to share the k8s secret?

Considering that we are on a operator, Nothing should prevent us from querying secrets in other namespaces (just like we are creating resources in other namespaces) given our rbac was given those permissions. wdyt?

Reply:
Ok, if we have the appropriate permissions to read secrets my approach would probably be to just read the secret defined in `passwordFromSecret` when creating dragonfly instances and use that as the admin/masterauth password.  I remember trying this initially and running into a permissions issue when reading the secret, but I'll give it another shot and see if I can get it working.  How does that sound?

Reply:
Sounds good! Did you have any blocker there @diffuse ? You can create the relevant rbac to get the operator access to the secret, and that should be it.

Reply:
Thanks!  No blocker, I just haven't had any time to get to this yet.  I should be able to implement it soon though.

Reply:
I finally found some time today and implemented most of this feature.  Once I run a few more tests I'll push it for review hopefully within the next 3 days.  Thanks for the patience!

Reply:
@Pothulapati I implemented the changes.  Let me know what you think.  Thanks!

Reply:
Thanks @diffuse! Taking a look today!

Reply:
Please let me know if any further explanation would help.